### PR TITLE
FB-1862: Implement Str() scalar string function

### PR DESCRIPTION
**Pull request by bruce-b-molecula**
STR() takes in a number and provides a string representation of it

first optional int variable will set a length to pad out to or to fill out with *'s if the number cannot be represented in that space (default 10)

Second optional variable will allow you to set the desired precision (default 0)

### CODE REVIEW

File: sql3/planner/inbuiltfunctionsstring.go [hunk 1]
```diff
@@ -644,3 +646,82 @@ func (n *callPlanExpression) EvaluateReplicate(currentRow []interface{}) (interf
...
+	}
+
+	coercedValue, err := coerceValue(n.args[0].Type(), parser.NewDataTypeDecimal(4), argOneEval, parser.Pos{})
+	if err != nil {
+		return "", err
+	}
+
+	decimalArgOne, ok := coercedValue.(pql.Decimal)
+	if !ok {
+		return nil, sql3.NewErrInternalf("unexpected type conversion '%T'", coercedValue)
```
Comment (HHans09) [inline sql3/planner/inbuiltfunctionsstring.go:687]: Once you rebase, we have new func 'NewErrUnexpectedTypeConversion' to wrap all the internal errors.

**Review by rachithrr — COMMENTED** [review]
Minor ones. Other than that, looks good to me.

**Comment by rachithrr** [comment]
Also, I think it would help if you can add the same commit message as the documentation for the function.

File: sql3/planner/inbuiltfunctionsstring.go [hunk 2]
```diff
@@ -783,3 +784,82 @@ func (n *callPlanExpression) EvaluateCharIndex(currentRow []interface{}) (interf
...
+		}
+	}
+
+	intArgThree := int64(0)
+	if len(n.args) > 2 {
+		argEval, err := n.args[2].Evaluate(currentRow)
+		if err != nil {
+			return nil, err
+		}
+		intArgThree, ok = argEval.(int64)
```
[the remark this answers is not in the document]
Comment (bruce-b-molecula) [reply sql3/planner/inbuiltfunctionsstring.go:849]: Added the null check for clarity

File: sql3/planner/inbuiltfunctionsstring.go [hunk 3]
```diff
@@ -783,3 +784,85 @@ func (n *callPlanExpression) EvaluateCharIndex(currentRow []interface{}) (interf
...
+		if !typeIsInteger(call.Args[i].DataType()) {
+			return nil, sql3.NewErrIntExpressionExpected(call.Args[i].Pos().Line, call.Args[i].Pos().Column)
+		}
+	}
+
+	call.ResultDataType = parser.NewDataTypeString()
+
+	return call, nil
+}
+
```
Comment (rachithrr) [inline sql3/planner/inbuiltfunctionsstring.go:811]: Can you add documentation for this?

**Review by rachithrr — COMMENTED** [review]
Looks good to me. Requires rebase

File: sql3/planner/inbuiltfunctionsstring.go (hunk 3 above)
Comment (bruce-b-molecula) [reply sql3/planner/inbuiltfunctionsstring.go:811]: Realized i was still connected to the old docs, moved it into a branch on the new docs
https://github.com/FeatureBaseDB/featurebase-docs/tree/fb-1862

**Review by rachithrr — APPROVED** [review]
lgtm

**Pull request merged**